fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -1271,7 +1314,27 @@ private void computeFragmentHosts() throws Exception {
                     FragmentExecParams param = fragmentExecParamsMap.get(fragment.getFragmentId());
                     int numBackends = param.scanRangeAssignment.size();
                     int numInstances = param.instanceExecParams.size();
-                    int pipelineDop = Math.max(1, degreeOfParallelism / Math.max(1, numInstances / numBackends));
+                    OlapScanNode scanNode = (OlapScanNode) leftMostNode;
+                    int numScanRangesPerInstance = scanNode.getScanRangeLocations(0).size() / numInstances;
+                    numScanRangesPerInstance = Math.max(1, numScanRangesPerInstance);
+                    long numRows = scanNode.getCardinality();
+                    if (scanNode.hasLimit()) {
+                        numRows = Math.min(numRows, scanNode.getLimit());
+                    }
+
+                    float inputBytes = numRows * scanNode.getAvgRowSize();
+                    long maxBytesPerOperator = ConnectContext.get().getSessionVariable().getPipelineMaxInputBytesPerOperator();
```

Comment:
Why here use input_bytes here? not row count?

Reply:
here, input_bytes is used to compute dop of the fragment instance,  input_bytes is more significant, for instance,  assuming a HashJoinScanNode follows the OlapScanNode, the larger avgRowSize, the more time to spend on HashJoinProbe computation.